### DISCUSSION TITLE
Installation option of Drive.

### DISCUSSION THREAD

**Discussion by joshyates1980** [nervosnetwork/neuron#2564] [Ideas]
For the executable on installing Neuron, will there be an option on future releases for the end user to select the drive/file location of the installation?

**Comment by Keith-CY** [comment]
> For the executable on installing Neuron, will there be an option on future releases for the end user to select the drive/file location of the installation?

Hi @joshyates1980 

I just went through the technical document roughly and found it's possible to specify the installation directory on windows. We'll have a try and deliver it in the next several versions if it works.

This feature could be tracked by https://github.com/Magickbase/neuron-public-issues/issues/92

Ref: 
- `allowToChangeInstallationDirectory`: https://www.electron.build/generated/nsisoptions

---

FYI, the most space-consuming directory, which includes data of ckb node, could be set in `Setting/Preference -> Data -> CKB Node Config & Storage`

![Preference](https://user-images.githubusercontent.com/7271329/210847938-b3ba23b9-63be-40b4-9323-8d6fadc46588.png)

**Reply by joshyates1980** [reply]
Thank you for the reply. I currently run the full node on my D: drive. It seems since I moved the full node from C: drive to D: drive during this last upgrade of ckb_v0.106.0_x86_64-pc-windows-msvc, my Neuron would not work on the C: drive.  Ultimately, I would enjoy to run both full node and neuron on my D: drive.


This would also be a nice feature, the option to move from C: drive to D: drive.

![image](https://user-images.githubusercontent.com/7255545/211027952-04835749-af2c-42be-89e0-55a001884383.png)

**Reply by EthAnalysis** [reply]
Full node and neuron both use the same ports by default. I couldn't start full node without turning Neuron off. You would have to adjust the port settings to make them both run on the same IP address.

**Reply by yanguoyu** [reply]
Hi, if you want to start an external ckb node, you can create a new network and switch to it. If we choose the `Internal Node`, the Neuron will start an internal ckb node.

<img width="1229" alt="image" src="https://github.com/nervosnetwork/neuron/assets/12881040/127b5b24-5da6-4a1b-bd22-1739debe0f1c">
